### DISCUSSION TITLE
.NET9 use of LocalizationResourceManager.Maui -> XamlC warning XC0103: Consider attributing the markup extension

### DISCUSSION THREAD

**Discussion by Pipozor** [SirJohnK/LocalizationResourceManager.Maui#38] [General]
Hello,

In my project, during build a warning appears, sample :

D:\Projects\LiDi\LiDi-lidi_dotnet9\LiDi.MobileApp\Pages\RankingsUsersPage.xaml(20,25): XamlC warning XC0103: Consider attributing the markup extension "LocalizationResourceManager.Maui.TranslateExtension" with [RequireService] or [AcceptEmptyServiceProvider] if it doesn't require any.

I tried to add :

`[RequireService([typeof(ILocalizationResourceManager)])]`

to C# code behind XAML but I still have the warning.

I'm wondering if this has to be added to your Markup extension code. What do you think ?

Like here : https://learn.microsoft.com/en-us/dotnet/maui/xaml/markup-extensions/create?view=net-maui-9.0

```
[RequireService([typeof(IReferenceProvider), typeof(IProvideValueTarget)])]
public class MyMarkupExtension : IMarkupExtension
{
    public object ProvideValue(IServiceProvider serviceProvider)
    {
        ...
        var referenceProvider = serviceProvider.GetService<IReferenceProvider>();
        var valueProvider = serviceProvider.GetService<IProvideValueTarget>() as IProvideParentValues
                                ?? throw new ArgumentException("serviceProvider does not provide an IProvideValueTarget");
        ...
    }
}

```

**Comment by SirJohnK** [comment]
Hi @Pipozor !

Yes, it needs to be added to the library. Will be changed in the upcoming 1.4.0.0 version. (See #37)

**Reply by Pipozor** [reply]
Thank you for your prompt answer. All the best.
